### ISSUE TITLE
Access Databases

### ISSUE THREAD

**Issue by FelipeCarrillo** [Kanaries/GWalkR#41]
This is not really an issue but is a question. 
I have large MS Access databases with over one million records. Does GWalkR support connections between these type of databases? If not, is there any plans to support this type of database in the future?

**Comment by ObservedObserver** [comment]
For now, I think you can connect your MS access db in [pygwalker](https://github.com/Kanaries/pygwalker) with [sqlalchemy-access](https://pypi.org/project/sqlalchemy-access/).

More docs about how to use connectors in pygwalker here: [doc](https://docs.kanaries.net/pygwalker/api-reference/dataset#working-with-databases)

**Comment by FelipeCarrillo** [comment]
I don't use Python, I am a heavily R user. 

Yahoo Mail: Search, Organize, Conquer 
 
  On Fri, Dec 20, 2024 at 6:17 PM, Elwynn ***@***.***> wrote:   


For now, I think you can connect your MS access db in pygwalker with sqlalchemy-access.

More docs about how to use connectors in pygwalker here: doc

**Comment by bruceyyu** [comment]
Yes, connecting to the database using DBI is on my to-do list! Please stay tuned and I will try to make it in early 2025

**Comment by FelipeCarrillo** [comment]
Great, thanks 

Yahoo Mail: Search, Organize, Conquer 
 
  On Wed, Dec 25, 2024 at 12:56 AM, ***@***.***> wrote:   


Yes, connecting to the database using DBI is on my to-do list! Please stay tuned and I will try to make it in early 2025

**Comment by FelipeCarrillo** [comment]
Any new developments on the access database request? Also, is there support for SQLite?

**Comment by gregvolny** [comment]
Any news  on accessing MySQL (maybe using DBI) in GwalkeR?

**Comment by bruceyyu** [comment]
Hi @FelipeCarrillo @gregvolny, thanks for your enquiries.
While developing this feature, I encountered difficulties in translating the DuckDB query to other database dialects in R, so the development was put on hold.
If you want to interact with your MySQL/SQLite databases visually, one workaround is to use [PyGWalker](https://github.com/Kanaries/pygwalker) in a Python environment, which has a mature DB access feature already.
Meanwhile, I'm still exploring possibilities to use [handy Python libraries](https://github.com/OHDSI/SqlRender) in R environment to do the dialect translations

**Comment by FelipeCarrillo** [comment]
👍
